From: Sam Macbeth <sam@cliqz.com>
Date: Fri, 11 Sep 2020 15:34:04 +0200
Subject: Remove tracking protection panel from url bar


---
diff --git a/browser/base/content/browser-init.js b/browser/base/content/browser-init.js
--- a/browser/base/content/browser-init.js
+++ b/browser/base/content/browser-init.js
@@ -558,7 +558,7 @@
     if (Services.prefs.getBoolPref("browser.search.widget.new", false)) {
       document.getElementById("searchbar-new")?.delayedStartupInit();
     }
-    gProtectionsHandler.init();
+    // gProtectionsHandler.init();
     gTrustPanelHandler.init();
 
     let safeMode = document.getElementById("helpSafeMode");
@@ -1219,7 +1219,7 @@
       Services.prefs.removeObserver(ctrlTab.prefName, ctrlTab);
       ctrlTab.uninit();
       gBrowserThumbnails.uninit();
-      gProtectionsHandler.uninit();
+      // gProtectionsHandler.uninit();
       gTrustPanelHandler.uninit();
       FullZoom.destroy();
 
diff --git a/browser/base/content/browser-siteIdentity.js b/browser/base/content/browser-siteIdentity.js
--- a/browser/base/content/browser-siteIdentity.js
+++ b/browser/base/content/browser-siteIdentity.js
@@ -981,10 +981,10 @@
     }
 
     // Hide the shield icon if it is a chrome page.
-    gProtectionsHandler._trackingProtectionIconContainer.classList.toggle(
-      "chromeUI",
-      this._isSecureInternalUI
-    );
+    // gProtectionsHandler._trackingProtectionIconContainer.classList.toggle(
+    //   "chromeUI",
+    //   this._isSecureInternalUI
+    // );
   },
 
   /**
diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -2073,7 +2073,7 @@
     const nsIWebProgressListener = Ci.nsIWebProgressListener;
 
     let browser = gBrowser.selectedBrowser;
-    gProtectionsHandler.onStateChange(aWebProgress, aStateFlags);
+    // gProtectionsHandler.onStateChange(aWebProgress, aStateFlags);
 
     if (
       aStateFlags & nsIWebProgressListener.STATE_START &&
@@ -2502,12 +2502,12 @@
       );
     }
 
-    gProtectionsHandler.onContentBlockingEvent(
-      aEvent,
-      aWebProgress,
-      aIsSimulated,
-      this._event // previous content blocking event
-    );
+    // gProtectionsHandler.onContentBlockingEvent(
+    //   aEvent,
+    //   aWebProgress,
+    //   aIsSimulated,
+    //   this._event // previous content blocking event
+    // );
 
     gTrustPanelHandler.onContentBlockingEvent(
       aEvent,
diff --git a/browser/base/content/browser.xhtml b/browser/base/content/browser.xhtml
--- a/browser/base/content/browser.xhtml
+++ b/browser/base/content/browser.xhtml
@@ -70,7 +70,7 @@
   <link rel="localization" href="browser/originControls.ftl"/>
   <link rel="localization" href="browser/panelUI.ftl"/>
   <link rel="localization" href="browser/places.ftl"/>
-  <link rel="localization" href="browser/protectionsPanel.ftl"/>
+  <!-- <link rel="localization" href="browser/protectionsPanel.ftl"/> -->
   <link rel="localization" href="browser/reportBrokenSite.ftl"/>
   <link rel="localization" href="browser/screenshots.ftl"/>
   <link rel="localization" href="browser/search.ftl"/>
