From: Sam Macbeth <sam@cliqz.com>
Date: Fri, 16 Oct 2020 15:47:47 +0200
Subject: Hide Protection Dashboard link from right burger menu

---
diff --git a/browser/base/content/appmenu-viewcache.inc.xhtml b/browser/base/content/appmenu-viewcache.inc.xhtml
--- a/browser/base/content/appmenu-viewcache.inc.xhtml
+++ b/browser/base/content/appmenu-viewcache.inc.xhtml
@@ -14,7 +14,7 @@
       <toolbaritem id="appMenu-menu-message"
                    closemenu="none">
       </toolbaritem>
-      <toolbaritem id="appMenu-fxa-status2"
+      <toolbaritem id="appMenu-fxa-status2" style="display: none !important;"
                    closemenu="none"
                    class="subviewbutton toolbaritem-combined-buttons">
         <html:div id="appMenu-fxa-text" data-l10n-id="appmenu-fxa-sync-and-save-data2"/>
diff --git a/browser/base/content/navigator-toolbox.inc.xhtml b/browser/base/content/navigator-toolbox.inc.xhtml
--- a/browser/base/content/navigator-toolbox.inc.xhtml
+++ b/browser/base/content/navigator-toolbox.inc.xhtml
@@ -220,7 +220,7 @@
 
           <!-- Use onclick instead of normal popup= syntax since the popup
                 code fires onmousedown, and hence eats our favicon drag events. -->
-          <box id="tracking-protection-icon-container" align="center"
+          <box id="tracking-protection-icon-container" align="center" style="display: none !important;"
                 role="button"
                 tooltip="tracking-protection-icon-tooltip"
                 urlbar-slot="site-info">
