From: Jenkins <jenkins@magrathea>
Date: Tue, 14 Feb 2023 20:32:16 +0100
Subject: Build: mach package use MOZ_FETCHES in automation

---
 build/moz.configure/bootstrap.configure | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/build/moz.configure/bootstrap.configure b/build/moz.configure/bootstrap.configure
index 646d83d340..8b08c00204 100644
--- a/build/moz.configure/bootstrap.configure
+++ b/build/moz.configure/bootstrap.configure
@@ -122,6 +122,7 @@ def bootstrap_path(path, **kwargs):
 
     @depends(
         "--enable-bootstrap",
+        "MOZ_AUTOMATION",
         want_bootstrap,
         toolchains_base_dir,
         moz_fetches_dir,
@@ -141,6 +142,7 @@ def bootstrap_path(path, **kwargs):
     @imports(_from="__builtin__", _import="Exception")
     def bootstrap_path(
         enable_bootstrap,
+        automation,
         want_bootstrap,
         toolchains_base_dir,
         moz_fetches_dir,
@@ -280,7 +282,7 @@ def bootstrap_path(path, **kwargs):
             try:
                 # With --enable-bootstrap=no-update, we don't `try_bootstrap`, except
                 # when the toolchain can't be found.
-                if (
+                if not automation and (
                     "no-update" not in enable_bootstrap or not exists
                 ) and not try_bootstrap(exists):
                     # If there aren't toolchain artifacts to use for this build,
-- 
2.52.0

