From d6ec2b481e1b79e65b8fb0aa99265c83ffa02a4f Mon Sep 17 00:00:00 2001
From: Jenkins <jenkins@magrathea>
Date: Sat, 21 Feb 2026 21:43:49 -0500
Subject: Disable newtab stories customize toggle and fix wallpaper
 updates

---
diff --git a/browser/extensions/newtab/content-src/components/Base/Base.jsx b/browser/extensions/newtab/content-src/components/Base/Base.jsx
--- a/browser/extensions/newtab/content-src/components/Base/Base.jsx
+++ b/browser/extensions/newtab/content-src/components/Base/Base.jsx
@@ -657,7 +657,10 @@
         wp => wp.title === selectedWallpaper
       );
       // solid color picker
-      if (selectedWallpaper.includes("solid-color-picker")) {
+      if (
+        typeof selectedWallpaper === "string" &&
+        selectedWallpaper.includes("solid-color-picker")
+      ) {
         const regexRGB = /#([a-fA-F0-9]{6})/;
         const hex = selectedWallpaper.match(regexRGB)?.[0];
         url = "";
diff --git a/browser/extensions/newtab/content-src/components/CustomizeMenu/ContentSection/ContentSection.jsx b/browser/extensions/newtab/content-src/components/CustomizeMenu/ContentSection/ContentSection.jsx
--- a/browser/extensions/newtab/content-src/components/CustomizeMenu/ContentSection/ContentSection.jsx
+++ b/browser/extensions/newtab/content-src/components/CustomizeMenu/ContentSection/ContentSection.jsx
@@ -5,7 +5,6 @@
 import React from "react";
 import { batch } from "react-redux";
 import { actionCreators as ac, actionTypes as at } from "common/Actions.mjs";
-import { SectionsMgmtPanel } from "../SectionsMgmtPanel/SectionsMgmtPanel";
 import { WallpaperCategories } from "../../WallpaperCategories/WallpaperCategories";
 // @nova-cleanup(move-directory): Update import path after WidgetsManagementPanel moves to components/CustomizeMenu/
 import { WidgetsManagementPanel } from "content-src/components/Nova/CustomizeMenu/WidgetsManagementPanel/WidgetsManagementPanel";
@@ -17,7 +16,6 @@
 
     // Refs are necessary for dynamically measuring drawer heights for slide animations
     this.topSitesDrawerRef = React.createRef();
-    this.pocketDrawerRef = React.createRef();
     this.widgetsMgmtDrawerRef = React.createRef();
   }
 
@@ -109,10 +107,19 @@
 
   componentDidMount() {
     this.setDrawerMargins();
-  }
-
-  componentDidUpdate() {
+    if (this.props.enabledSections.pocketEnabled) {
+      this.props.setPref("feeds.section.topstories", false);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
     this.setDrawerMargins();
+    if (
+      this.props.enabledSections.pocketEnabled &&
+      !prevProps.enabledSections.pocketEnabled
+    ) {
+      this.props.setPref("feeds.section.topstories", false);
+    }
   }
 
   setDrawerMargins() {
@@ -120,10 +127,6 @@
       `TOP_SITES`,
       this.props.enabledSections.topSitesEnabled
     );
-    this.setDrawerMargin(
-      `TOP_STORIES`,
-      this.props.enabledSections.pocketEnabled
-    );
     this.setDrawerMargin(`WIDGETS`, this.props.widgetsEnabled);
   }
 
@@ -133,9 +136,6 @@
     switch (drawerID) {
       case `TOP_SITES`:
         drawerRef = this.topSitesDrawerRef.current;
-        break;
-      case `TOP_STORIES`:
-        drawerRef = this.pocketDrawerRef.current;
         break;
       case `WIDGETS`:
         drawerRef = this.widgetsMgmtDrawerRef.current;
@@ -164,8 +164,6 @@
     const {
       enabledSections,
       enabledWidgets,
-      pocketRegion,
-      mayHaveInferredPersonalization,
       mayHaveWeather,
       mayHaveWidgets,
       mayHaveTimerWidget,
@@ -177,12 +175,9 @@
       wallpapersUserEnabled,
       activeWallpaper,
       setPref,
-      mayHaveTopicSections,
       weatherDisplay,
       exitEventFired,
       onSubpanelToggle,
-      toggleSectionsMgmtPanel,
-      showSectionsMgmtPanel,
       // @nova-cleanup(remove-conditional): Remove novaEnabled
       novaEnabled,
       wallpapersEnabled,
@@ -192,22 +187,10 @@
     } = this.props;
     const {
       topSitesEnabled,
-      pocketEnabled,
       weatherEnabled,
-      showInferredPersonalizationEnabled,
       topSitesRowsCount,
     } = enabledSections;
     const { timerEnabled, listsEnabled, clocksEnabled } = enabledWidgets;
-
-    // @nova-cleanup(remove-conditional): Remove novaEnabled check and newtab-custom-stories-toggle, default to newtab-recommended-stories-toggle
-    let pocketToggleL10nId;
-    if (mayHaveInferredPersonalization) {
-      pocketToggleL10nId = "newtab-custom-stories-personalized-toggle";
-    } else if (novaEnabled) {
-      pocketToggleL10nId = "newtab-recommended-stories-toggle";
-    } else {
-      pocketToggleL10nId = "newtab-custom-stories-toggle";
-    }
 
     // @nova-cleanup(remove-conditional): This conditional adds the toggle for wallpaper visibility.
     return (
@@ -380,115 +363,6 @@
               </moz-toggle>
             </div>
 
-            {
-              // @nova-cleanup(remove-conditional): Remove novaEnabled check, keep divider
-              novaEnabled && mayHaveWidgets && (
-                <span className="divider" role="separator"></span>
-              )
-            }
-            {
-              // @nova-cleanup(remove-conditional): Remove novaEnabled check, keep toggle and WidgetsManagementPanel
-              novaEnabled && mayHaveWidgets && (
-                <div id="widgets-section" className="section">
-                  <moz-toggle
-                    id="widgets-system-toggle"
-                    pressed={widgetsEnabled || null}
-                    ontoggle={this.onPreferenceSelect}
-                    data-preference="widgets.enabled"
-                    data-event-source="WIDGETS_SYSTEM"
-                    data-l10n-id="newtab-custom-widget-section-toggle"
-                  >
-                    <div slot="nested">
-                      <div className="more-info-widgets-wrapper">
-                        <div
-                          className="more-information"
-                          ref={this.widgetsMgmtDrawerRef}
-                        >
-                          <WidgetsManagementPanel
-                            enabledSections={enabledSections}
-                            enabledWidgets={enabledWidgets}
-                            mayHaveWeather={mayHaveWeather}
-                            mayHaveTimerWidget={mayHaveTimerWidget}
-                            mayHaveListsWidget={mayHaveListsWidget}
-                            mayHaveSportsWidget={mayHaveSportsWidget}
-                            mayHaveClocksWidget={mayHaveClocksWidget}
-                            mayHaveWeatherForecast={mayHaveWeatherForecast}
-                            weatherDisplay={weatherDisplay}
-                            setPref={setPref}
-                            onSubpanelToggle={onSubpanelToggle}
-                            togglePanel={toggleWidgetsManagementPanel}
-                            showPanel={showWidgetsManagementPanel}
-                          />
-                        </div>
-                      </div>
-                    </div>
-                  </moz-toggle>
-                </div>
-              )
-            }
-
-            {
-              // @nova-cleanup(remove-conditional): Remove novaEnabled check, keep divider
-              // The pocketRegion check makes sure there is only one divider present if it's false
-              novaEnabled && pocketRegion && (
-                <span className="divider" role="separator"></span>
-              )
-            }
-
-            {pocketRegion && (
-              <div id="pocket-section" className="section">
-                <moz-toggle
-                  id="pocket-toggle"
-                  pressed={pocketEnabled || null}
-                  ontoggle={this.onPreferenceSelect}
-                  data-preference="feeds.section.topstories"
-                  data-event-source="TOP_STORIES"
-                  data-l10n-id={pocketToggleL10nId}
-                >
-                  <div slot="nested">
-                    {(mayHaveInferredPersonalization ||
-                      mayHaveTopicSections) && (
-                      <div className="more-info-pocket-wrapper">
-                        <div
-                          className="more-information"
-                          ref={this.pocketDrawerRef}
-                        >
-                          {mayHaveInferredPersonalization && (
-                            <div className="check-wrapper" role="presentation">
-                              <input
-                                id="inferred-personalization"
-                                className="customize-menu-checkbox"
-                                disabled={!pocketEnabled}
-                                checked={showInferredPersonalizationEnabled}
-                                type="checkbox"
-                                onChange={this.onPreferenceSelect}
-                                data-preference="discoverystream.sections.personalization.inferred.user.enabled"
-                                data-event-source="INFERRED_PERSONALIZATION"
-                              />
-                              <label
-                                className="customize-menu-checkbox-label"
-                                htmlFor="inferred-personalization"
-                                data-l10n-id="newtab-custom-stories-personalized-checkbox-label"
-                              />
-                            </div>
-                          )}
-                          {mayHaveTopicSections && (
-                            <SectionsMgmtPanel
-                              exitEventFired={exitEventFired}
-                              pocketEnabled={pocketEnabled}
-                              onSubpanelToggle={onSubpanelToggle}
-                              togglePanel={toggleSectionsMgmtPanel}
-                              showPanel={showSectionsMgmtPanel}
-                              novaEnabled={novaEnabled}
-                            />
-                          )}
-                        </div>
-                      </div>
-                    )}
-                  </div>
-                </moz-toggle>
-              </div>
-            )}
           </div>
           {
             // @nova-cleanup(remove-conditional): Remove this divider once Nova lands
diff --git a/browser/extensions/newtab/content-src/components/WallpaperCategories/WallpaperCategories.jsx b/browser/extensions/newtab/content-src/components/WallpaperCategories/WallpaperCategories.jsx
--- a/browser/extensions/newtab/content-src/components/WallpaperCategories/WallpaperCategories.jsx
+++ b/browser/extensions/newtab/content-src/components/WallpaperCategories/WallpaperCategories.jsx
@@ -487,10 +487,13 @@
     const selectedWallpaper = prefs["newtabWallpapers.wallpaper"];
 
     // User has previous selected a custom color
-    if (selectedWallpaper.includes("solid-color-picker")) {
+    if (
+      typeof selectedWallpaper === "string" &&
+      selectedWallpaper.includes("solid-color-picker")
+    ) {
       showColorPicker = true;
       const regex = /#([a-fA-F0-9]{6})/;
-      [wallpaperCustomSolidColorHex] = selectedWallpaper.match(regex);
+      wallpaperCustomSolidColorHex = selectedWallpaper.match(regex)?.[0] || null;
     }
 
     // Remove last item of solid colors to make space for custom color picker
