From b8f024a59c72c50c5a896b42df9a8c0131db3321 Mon Sep 17 00:00:00 2001
From: Jenkins <jenkins@magrathea>
Date: Sat, 21 Feb 2026 21:50:40 -0500
Subject: Remove recommended stories from Home preferences UI

---
diff --git a/browser/components/preferences/home.inc.xhtml b/browser/components/preferences/home.inc.xhtml
--- a/browser/components/preferences/home.inc.xhtml
+++ b/browser/components/preferences/home.inc.xhtml
@@ -120,7 +120,6 @@
   <vbox id="web-search" />
   <vbox id="weather" />
   <vbox id="topsites" />
-  <vbox id="topstories" />
   <vbox id="support-firefox" />
 
   <html:moz-box-item class="mission-message">
diff --git a/browser/extensions/newtab/lib/AboutPreferences.sys.mjs b/browser/extensions/newtab/lib/AboutPreferences.sys.mjs
--- a/browser/extensions/newtab/lib/AboutPreferences.sys.mjs
+++ b/browser/extensions/newtab/lib/AboutPreferences.sys.mjs
@@ -83,23 +83,6 @@
       eventSource: "TOP_SITES",
     },
     {
-      id: "topstories",
-      pref: {
-        feed: "feeds.section.topstories",
-        titleString: {
-          id: "home-prefs-recommended-by-header-generic",
-        },
-        descString: {
-          id: "home-prefs-recommended-by-description-generic",
-        },
-      },
-      shouldHidePref: !Services.prefs.getBoolPref(
-        "browser.newtabpage.activity-stream.feeds.system.topstories",
-        true
-      ),
-      eventSource: "TOP_STORIES",
-    },
-    {
       id: "support-firefox",
       pref: {
         feed: "showSponsoredCheckboxes",
@@ -109,19 +92,6 @@
             name: "showSponsoredTopSites",
             titleString: "home-prefs-shortcuts-by-option-sponsored",
             eventSource: "SPONSORED_TOP_SITES",
-          },
-          {
-            name: "showSponsored",
-            titleString: "home-prefs-recommended-by-option-sponsored-stories",
-            eventSource: "POCKET_SPOCS",
-            shouldHidePref: !Services.prefs.getBoolPref(
-              "browser.newtabpage.activity-stream.feeds.system.topstories",
-              true
-            ),
-            shouldDisablePref: !Services.prefs.getBoolPref(
-              "browser.newtabpage.activity-stream.feeds.section.topstories",
-              true
-            ),
           },
         ],
       },
@@ -1881,7 +1851,6 @@
         });
       }
 
-      setupSupportFirefoxSubCheck("feeds.section.topstories", "showSponsored");
       setupSupportFirefoxSubCheck("feeds.topsites", "showSponsoredTopSites");
     }
 
