From 87bfc88f5f5f6eeceb7b68dce49ebc7fcc1ef820 Mon Sep 17 00:00:00 2001
From: Jenkins <jenkins@magrathea>
Date: Sat, 21 Feb 2026 22:05:40 -0500
Subject: [PATCH] Sync newtab bundle with stories removal and wallpaper fixes

---
diff --git a/browser/extensions/newtab/data/content/activity-stream.bundle.js b/browser/extensions/newtab/data/content/activity-stream.bundle.js
--- a/browser/extensions/newtab/data/content/activity-stream.bundle.js
+++ b/browser/extensions/newtab/data/content/activity-stream.bundle.js
@@ -22503,10 +22503,10 @@
     const selectedWallpaper = prefs["newtabWallpapers.wallpaper"];
 
     // User has previous selected a custom color
-    if (selectedWallpaper.includes("solid-color-picker")) {
+    if (typeof selectedWallpaper === "string" && selectedWallpaper.includes("solid-color-picker")) {
       showColorPicker = true;
       const regex = /#([a-fA-F0-9]{6})/;
-      [wallpaperCustomSolidColorHex] = selectedWallpaper.match(regex);
+      wallpaperCustomSolidColorHex = selectedWallpaper.match(regex)?.[0] || null;
     }
 
     // Remove last item of solid colors to make space for custom color picker
@@ -23037,9 +23037,15 @@
   }
   componentDidMount() {
     this.setDrawerMargins();
-  }
-  componentDidUpdate() {
+    if (this.props.enabledSections.pocketEnabled) {
+      this.props.setPref("feeds.section.topstories", false);
+    }
+  }
+  componentDidUpdate(prevProps) {
     this.setDrawerMargins();
+    if (this.props.enabledSections.pocketEnabled && !prevProps.enabledSections.pocketEnabled) {
+      this.props.setPref("feeds.section.topstories", false);
+    }
   }
   setDrawerMargins() {
     this.setDrawerMargin(`TOP_SITES`, this.props.enabledSections.topSitesEnabled);
@@ -23289,10 +23295,10 @@
     })))))),
     // @nova-cleanup(remove-conditional): Remove novaEnabled check, keep divider
     // The pocketRegion check makes sure there is only one divider present if it's false
-    novaEnabled && pocketRegion && /*#__PURE__*/external_React_default().createElement("span", {
+    novaEnabled && false && /*#__PURE__*/external_React_default().createElement("span", {
       className: "divider",
       role: "separator"
-    }), pocketRegion && /*#__PURE__*/external_React_default().createElement("div", {
+    }), false && /*#__PURE__*/external_React_default().createElement("div", {
       id: "pocket-section",
       className: "section"
     }, /*#__PURE__*/external_React_default().createElement("moz-toggle", {
@@ -26610,7 +26616,7 @@
     } else if (wallpaperList) {
       const wallpaper = wallpaperList.find(wp => wp.title === selectedWallpaper);
       // solid color picker
-      if (selectedWallpaper.includes("solid-color-picker")) {
+      if (typeof selectedWallpaper === "string" && selectedWallpaper.includes("solid-color-picker")) {
         const regexRGB = /#([a-fA-F0-9]{6})/;
         const hex = selectedWallpaper.match(regexRGB)?.[0];
         url = "";
